File: src/main/java/com/server/ordering/service/MemberService.java
package com.server.ordering.service;

import com.server.ordering.domain.dto.request.SignInDto;
import com.server.ordering.domain.member.MemberBase;

import java.util.Optional;

public interface MemberService<T extends MemberBase> {

    Long signUp(Object signUpDto);
    T signIn(SignInDto signInDto);
    void signOut(Long id);
    boolean isIdDuplicated(String signInId);
    void deleteAccount(Long id);
}

File: src/main/kotlin/com/byeoru/ordering_server/service/MemberService.kt
package com.byeoru.ordering_server.service

import com.byeoru.ordering_server.domain.dto.request.SignInDto
import com.byeoru.ordering_server.domain.member.MemberBase

interface MemberService<T : MemberBase> {
    fun signUp(signUpDto: Any): Long?
    fun signIn(signInDto: SignInDto): T?
    fun signOut(id: Long)
    fun isIdDuplicated(signInId: String): Boolean
    fun deleteAccount(id: Long)
}